---
title: "Final Project"
author: "Shannon Baker"
mainfont: DejaVu Sans
output: html_document

---
[Final Project R Script](https://github.com/ShannonBa/Data_Course_BAKER/blob/main/Final_Project/Final_Project_script.R)

<br/>

### Introduction and Background  

My final project reviews how plots were created in the the research article "[Reprogramming to recover youthful epigenetic information and restore vision](https://rdcu.be/ch1re)" by Lu et al.[*](#anchor) The paper analyzed what genes in mice are affected in aging and how these genes can be changed to reverse aging in mice. 

For the experiment, mice eyes are looked at focusing on retinal ganglion cells. The genes __Oct4, Sox2, and Klf4 genes (OSK)__ were introduced with a dual adeno-associated virus (AAV). The OSK genes helped with axon regeneration by causing more expression of DNA demethylases TET1 and TET2. Due to The TET _demtheylase_ activity, _more_ genes were expressed in older mice because these genes were demethylated and open to transcription again. This demethylation allowed these suppressed genes to become more active in older mice, causing the gene expression to resemble younger mice gene expression.  

The two figures being recreated compare the gene expression of young mice to old mice to OSK expressing mice. In older OSK mice, there is gene expression similar to young mice. This increased gene expression helps older mice with axon regeneration after injury and reverses vision loss in a mouse model of glaucoma and in aged mice.



<br/>

```{r include=FALSE}
## load packages
library(tidyverse)
library(janitor)
library(dplyr)
library(gplots)
library(ggplot2)
library(viridis)
library(RColorBrewer)
library(kableExtra)
```

### Images recreated from Fig. 4: 
  
  
#### Heatmap  
<p align="left">
 ![Caption](./heatmap.PNG)
 </p>
This heatmap shows the genes that are actively expressed in young mice and those that are less active in old mice. The darker the color, the less that gene is expressed. There is a lot of darker color in the old mice that is due to more methylation inhibiting gene expression. 

<br/>

#### Volcano plot  
<p align="left">
 ![](./foldchange.PNG)
 </p>
This volcano plot shows the genes that are significantly active in young mice, those genes that have decreased expression in old mice, and those genes that are increased in old mice with the OSK genes being expressed. The log2 is showing a fold change. For every fold change of 1, the expression is doubling. So, on the x-axis right side of the graph, this is showing genes that have increased expression in young mice. The left side is decreased gene expression in old mice. For the y-axis part of the graph, OSK expression is shown to increase gene expression in old mice (up higher) over old mice that do not have the induced OSK expression (down lower). 





<br/>

### Re-creation  
 From the paper, there was no RStudio code available, but the data used to create the two graphs was available. The data for both the heatmap and the volcano plot were together, but the data was formatted for optimal use in RStudio. 
```{r echo=FALSE, message=FALSE, warning=FALSE}
### load data

df <- read_csv("./41586_2020_2975_MOESM11_ESM.csv", skip=3)
names(df) <- names(df) %>% str_replace_all(" ","_")
#glimpse(df)
skimr::skim_without_charts(df)
#or kablextraa
```
<br/>
There are 464 genes analyzed for five 5-month old mice, six 12-month old mice with no OSK AAV added, five 12-month old mice with an AAV added but with no OSK genes expressed, and four 12-month old mice with the OSK genes being expressed. 


```{r include=FALSE}
heatdf <- df[ ,c(                
  "5_mo_rep2"  ,                      
  "5_mo_rep3"   ,                     
  "5_mo_rep4"    ,                    
  "5_mo_rep5"     ,                   
  "12_mo_rep1" ,                      
  "12_mo_rep2"  ,                     
  "12_mo_rep3"  ,                     
  "12_mo_rep4"   ,                    
  "12_mo_rep5"    ,                   
  "12_mo_rep6"     ,                  
  "12_mo_(-OSK)_rep1",                
  "12_mo_(-OSK)_rep2" ,               
  "12_mo_(-OSK)_rep3"  ,              
  "12_mo_(-OSK)_rep4"   ,             
  "12_mo_(-OSK)_rep5"    ,            
  "12_mo_(+OSK)_rep1"     ,           
  "12_mo_(+OSK)_rep2" ,               
  "12_mo_(+OSK)_rep3"  ,              
  "12_mo(+OSK)_rep4")]
```


<br/>


### Heatmap created
This heatmap is comparing gene expression in young, old, and OSK old mice.   
  
For the colors, the lighter, more yellow the color, the more active the gene is in that mouse. The darker, more purple the color, the less that gene is expressed in that mouse. Dark purple to black means that gene is not being expressed.   
  
The colored bar at the top is showing the different groups of mice. The __black__ line marks the four columns in the heatmap that represent the genes active in __5 month old mice__. __Grey__ represents __12 month old mice with no treatment__. __Red__ represents __12 month old mice with AVV treatment but no OSK expression__. __Blue__ represents __12 month old mice with AVV treatment to induce expression of the OSK genes__.  
  

```{r echo=FALSE}
#### Making heatmap ####
heat_matrix <- as(heatdf, "matrix")
rownames(heat_matrix) <- df$gene

#color coding
mousenames <- c(rep("Young",4),
                rep("Old",6),
                rep("Old(-OSK)",5),
                rep("Old(+OSK)",4))
mousecolors <- plyr::mapvalues(mousenames,
                               from = unique(mousenames),
                               to=c("Black","Gray","Red","Blue"))
gene_colors <- viridis::magma(dim(heat_matrix)[1])



#actual heatmap
heatmap(heat_matrix, Colv = NA,
        ColSideColors = mousecolors,
        margins = c(10,10),
        col=gene_colors)
```
<br/>

### Volcano plot created  
The volcano plot was made from the q-alues, as the p-value can give the expected false positive rate obtained by rejecting the null hypothesis for any result with an equal or smaller p-value, the q-value gives the expected pFDR obtained by rejecting the null hypothesis for any result with an equal or smaller q-value.  
  
```{r echo=FALSE}
df %>%
  mutate(Color = case_when(QValue < 0.05 & QValue_2 > 0.05  ~ "red",
                           QValue > 0.05 & QValue_2 < 0.05 & QValue_1 < 0.05  ~ "Significant in both",
                           QValue < 0.05 & QValue_1 > 0.05  ~ "unknown1",
                           QValue < 0.05 & QValue_2 < 0.05  ~ "Significant in both (sensory)",
                           QValue > 0.05 & QValue_2 > 0.05  ~ "Significant in ageing",
                           QValue > 0.05 & QValue_1 < 0.05  ~ "Significant in both",
                           QValue > 0.05 & QValue_1 > 0.05  ~ "unknown2"))%>%
  ggplot(aes(x=`logFC_(12m/5m)`, y=`logFC_(12_mo_(+OSK)/12_mo_(-OSK))`, color=Color)) + geom_point() +
  scale_color_manual(values = c("grey", "blue", "red","green","yellow","black","brown")) +
  geom_hline(yintercept=0, color= "grey", size= 1.0) +
  geom_vline(xintercept = 0, color= "grey", size= 1.0) +
  geom_abline(intercept = 0, slope = -1, color="grey", 
              linetype="dashed", size=1.0) +
  geom_abline(intercept = 0, slope = 1, color="grey", 
              linetype="dashed", size=1.0)+
  theme(panel.grid.major = element_blank(), panel.grid.minor = element_blank(),
        panel.background = element_blank(), axis.line = element_line(colour = "black"))+
  labs( x ="log2(Old/Young)", y = "log2(+OSK/-OSK)") +
  annotate("text", x=-1.5, y=-2, label= "Down in ageing") + 
  annotate("segment", x = 0, xend = -1.5, y = -3, yend = -3, colour = "#ffce42", size=1, alpha=1, arrow=arrow(length = unit(0.1, "inches"), ends = "last", type = "closed"))+
  annotate("text", x=-6, y=-.25, label= "Up by OSK") + 
  annotate("segment", x = -6, xend = -6, y = 0, yend = 1, colour = "#ffce42", size=1, alpha=1, arrow=arrow(length = unit(0.1, "inches"), ends = "last", type = "closed"))


```
  
<br/>

### Conclusion  

My heatmap and volcano plot lack certain features that the paper acquired, such as a different grouping of gene for the dendrogram of the heatmap and significance points of both old and upregulated sensory genes in mice for the volcano plot, but with the data provided, it makes sense how the authors Lu et al.[*](#anchor) came to their conclusions of OSK having a significant affect on old mice gene expression. In the paper, the researchers were trying to find a way to help older mice acquire the traits of younger mice to help with injury and diseases related to the eye such as glaucoma. From their experiment, they have shown that the genes OSK can have an affect on triggering demthylation of DNA and returning a cell to a phenotype more similar to that of a young mouse in a beneficial way.   

<br/>

### Article Reference: {#anchor}  
Lu, Y., Brommer, B., Tian, X. et al. Reprogramming to recover youthful epigenetic information and restore vision. Nature 588, 124â€“129 (2020). https://doi.org/10.1038/s41586-020-2975-4


___

[Home](https://shannonba.github.io/index.html) | [About me](https://shannonba.github.io/AboutMe) | [Final Project](https://shannonba.github.io/Final_Project) | [Hobbies](https://shannonba.github.io/Hobbies)

<style>
  body {
    background-color: #FFEABF;
    margin-top: 50px;
  }

 /*Navigation Panel*/
  /*Navigation panel position - ul, li, li a, and .active*/
    ul{
      list-style-type: none;
      margin: 0; /*Top Margine*/
      padding: 0; /*Padding of HOME from the edge*/
      overflow: hidden;
      background-color: #333;
      position: fixed;
      top: 0; /*Does Nothing*/
      width: 100%; /*Does Nothing*/
      }
    ul {
        position: fixed;
        top: 0; /*Does Nothing*/
        width: 84%; /*Adjusts the width of the background on the NAVIGATION BAR.*/
      }
    li {
        float: left;
      }
    li a {
        display: block;
        color: white;
        text-align: center;
        padding: 14px 16px;
        text-decoration: none;
      }
    li a:hover:not(.active) {
        background-color: #111;
      }
    .active {
        background-color: #F0BB53;
      }
</style>

  <ul>
    <li><a href="index.html">Home</a></li>
    <li><a href="AboutMe.html">About Me</a></li>
    <li><a class="active" href="Final_Project.html">Final Project</a></li>
     <li><a href="Hobbies.html">Hobbies</a></li>
  </ul>